src/frontend/screens/ManualGpsScreen/shared.ts

```diff
@@ -0,0 +1,36 @@
+export type CoordinateField = "lat" | "lon";
+
+type Coordinates = { lat?: number | null; lon?: number | null };
+
+export type ConvertedCoordinateData = {
+  coords?: Coordinates;
+  error?: Error;
+};
+
+export type FormProps = {
+  coords?: Coordinates;
+  onValueUpdate: (convertedCoordinates: ConvertedCoordinateData) => void;
```

Comment:
@achou11 Im unsure as to what you are doing here with `onValueUpdate` Prop

Reply:
this is used as a format-agnostic state updater for the `ManualGpsScreen` that's used by any of the forms within it (`utm`, `dd`, `dms`). The idea is that we keep a top-level state for the end coordinates that we want to eventually save (lat + lon) but we need to be able to update that state in any of the forms. The forms themselves handle their own coordinate state based on the specifics of their format (including string formatting and whatnot), and then whenever valid input changes occur, we call this callback so that the global draft state is updated with the converted coordinates (e.g. UTM form will have UTM-specific input changes occur, then if the values are valid, we convert them to lat+lon and provide that in this callback)